python-package/lightgbm/basic.py

```diff
@@ -3159,8 +3165,8 @@ def eval(self, data, name, feval=None):
                 is_higher_better : bool
                     Is eval result higher better, e.g. AUC is ``is_higher_better``.
 
-            For multi-class task, the preds is group by class_id first, then group by row_id.
-            If you want to get i-th row preds in j-th class, the access way is preds[j * num_data + i].
+            For multi-class task, preds are a [n_samples, n_classes] numpy 2-D array,
```

Comment:
Could you please also check that customized evaluation function with multi class works correctly? I've read the code, and it seems that the customized evaluation function will finally take the output of `__inner_predict` as input, which is of the shape `n_sample * n_class`. This is inconsistent with the hint here.
https://github.com/microsoft/LightGBM/blob/820ae7e65e90860061ec89cc4a1ad0a39b96e051/python-package/lightgbm/basic.py#L3840

Reply:
Hmm you're right. I've only modified the portions required for `fobj`, I'll work on `feval`.